README.md

```diff
@@ -1,207 +1,607 @@
+| href              | string                                    | URI to the model artifact.                                                                       |
+| type              | string                                    | The media type of the artifact (see [Model Artifact Media-Type](#model-artifact-media-type).     |
+| roles             | \[string]                                 | **REQUIRED** Specify `mlm:model`. Can include `["mlm:weights", "mlm:checkpoint"]` as applicable. |
+| mlm:artifact_type | [Artifact Type Enum](#artifact-type-enum) | Specifies the kind of model artifact. Typically related to a particular ML framework.            |
+
+Recommended Asset `roles` include `mlm:weights` or `mlm:checkpoint` for model weights that need to be loaded by a
+model definition and `mlm:compiled` for models that can be loaded directly without an intermediate model definition.
+In each case, the `mlm:model` should be applied as well to indicate that this asset represents the model.
+
+It is also recommended to make use of the
+[file](https://github.com/stac-extensions/file?tab=readme-ov-file#asset--link-object-fields)
+extension for this Asset, as it can provide useful information to validate the contents of the model definition,
+by comparison with fields `file:checksum` and `file:size` for example.
+
+#### Model Artifact Media-Type
+
+Not all ML framework, libraries or model artifacts provide explicit media-type. When those are not provided, custom
+media-types can be considered. For example `application/x-pytorch` or `application/octet-stream; application=pytorch`
+could be appropriate to represent a PyTorch `.pt` file, since the underlying format is a serialized pickle structure.
+
+#### Artifact Type Enum
+
+This value can be used to provide additional details about the specific model artifact being described.
+For example, PyTorch offers various strategies for providing model definitions, such as Pickle (`.pt`), TorchScript,
+or the compiled approach. Since they all refer to the same ML framework,
+the [Model Artifact Media-Type](#model-artifact-media-type) would be insufficient in this case to detect with strategy
+should be used. 
+
+Following are some proposed *Artifact Type* values for corresponding approaches, but other names are
+permitted as well. Note that the names are selected using the framework-specific definitions to help
+the users understand the source explicitly, although this is not strictly required either.
+
+| Artifact Type      | Description                                                                                                              |
+|--------------------|--------------------------------------------------------------------------------------------------------------------------|
+| `torch.compile`    | A model artifact obtained by [`torch.compile`](https://pytorch.org/tutorials/intermediate/torch_compile_tutorial.html).  |
```

Comment:
```suggestion
| `.pt2`    | A model artifact obtained using [Pytorch's AOTInductor](https://pytorch.org/docs/main/torch.compiler_aot_inductor.html).  |
```

I think this is a necessary edit since torch.compile is an API for compiling pytorch nn.Modules. it's used to speed up Pytorch code in general, for training or inference. Many of the backend internals that make torch.compile work are used in AOTInductor (the tool that creates compiled model artifacts) but they aren't the same thing and I think here we want to refer to artifacts produced by AOTInductor

Reply:
I've added some reference mentions of AOTInductor and `torch.export`. Let me know what you think of the edit.

I must say, though, I'm not 100% convinced about these definitions included in the spec (or rather, how explicit to be about them). They are still fairly prone to change soon. Even the doc mentions that many terms are used interchangeably. 

https://pytorch.org/docs/main/torch.compiler.html#torch-compiler
> In some cases, the terms torch.compile, TorchDynamo, torch.compiler might be used interchangeably in this documentation.

There is also a lot of ambiguity about `torch.export` using TorchDynamo under the hood, but then, `torch.compile` accepts `backend='inductor'` for TorchInductor (https://pytorch.org/docs/main/torch.compiler.html#torch-compiler), although they are also listed as distinct technologies under `torch.compiler`. And then, there is `torch.export.save`/`torch.export.save` for `.pt2` (https://pytorch.org/docs/main/export.html#serialization) as parallel to the `torch.save`/`torch.load` for `.pt`.

I do not want to make the maintenance of the spec a burden for any new subtle change from pytorch. It's a fast-evolving ecosystem. Since this is an open field, I fear it is getting slightly too specific for this framework.